### PR TITLE
🐛Source Quickbooks: Fix configured catalog for discovering schemas with new cdk

### LINKED ISSUE #40596: [source-quickbooks] discovered schema is empty
### Connector Name

source-quickbooks

### Connector Version

3.0.4-3.0.7

### What step the error happened?

Updating the connector

### Relevant information

After upgrading quickbooks connector from 3.0.3 to 3.0.4 (and newer) all table schemas are gone and Airbyte tries to remove all fields. I reviewed the source code changes and there doesn't seem to be any changes in the connector code, but just library updates (airbyte_cdk from 0.63 to 1.0).

When I run discover command on the `airbyte/source-quickbooks:3.0.4` docker image (and newer) or using poetry locally it returns `"json_schema": {}` for all streams, but when I run it on `airbyte/source-quickbooks:3.0.3` it returns correct schema. 

I noticed that upgrading airbyte_cdk to 1.7.1 fixes the schema discovery but it' incompatible with custom connector components. Can somebody look into it?

### Relevant log output

```shell
✗ poetry run source-quickbooks discover --config secrets/config.json
The currently activated Python version 3.12.3 is not supported by the project (^3.9,<3.12).
Trying to find and use a compatible version. 
Using python3 (3.10.14)
{"type": "CATALOG", "catalog": {"streams": [{"name": "accounts", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "bill_payments", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "bills", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "budgets", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "classes", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "credit_memos", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "customers", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "departments", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "deposits", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "employees", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "estimates", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "invoices", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "items", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "journal_entries", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "payment_methods", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "payments", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "purchase_orders", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "purchases", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "refund_receipts", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "sales_receipts", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "tax_agencies", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "tax_codes", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "tax_rates", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "terms", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "time_activities", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "transfers", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "vendor_credits", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "vendors", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["airbyte_cursor"], "source_defined_primary_key": [["Id"]]}, {"name": "profit_and_loss", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["end_date"], "source_defined_primary_key": [["start_date"], ["end_date"]]}, {"name": "profit_and_loss_rows", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["end_date"], "source_defined_primary_key": [["start_date"], ["end_date"], ["section"], ["account"], ["account_id"]]}, {"name": "balance_sheet", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["end_date"], "source_defined_primary_key": [["start_date"], ["end_date"]]}, {"name": "balance_sheet_rows", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["end_date"], "source_defined_primary_key": [["start_date"], ["end_date"], ["section"], ["account"], ["account_id"]]}, {"name": "trial_balance", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["end_date"], "source_defined_primary_key": [["start_date"], ["end_date"]]}, {"name": "trial_balance_rows", "json_schema": {}, "supported_sync_modes": ["full_refresh", "incremental"], "source_defined_cursor": true, "default_cursor_field": ["end_date"], "source_defined_primary_key": [["start_date"], ["end_date"], ["section"], ["account"], ["account_id"]]}]}}
```

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by btkcodedev**
Closes https://github.com/airbytehq/airbyte/issues/40596

- The new CDK version introduced problems with discovering schemas due to external JSON files
- Reverted the deletion of configured catalog to support new CDK
- Chore: Deleted requirements.txt and moved spec inline

### CODE REVIEW

**Comment by btkcodedev** [comment]
Hopefully this will fix the issue, Asked for GSM creds to verify the read command

**Comment by natikgadzhi** [comment]
@DanyloGL in touch with you re: creds?

**Comment by natikgadzhi** [comment]
> Reverted the deletion of configured catalog to support new CDK

Weird that CDK update requires a catalog. Hmm. Did you run CATs locally? How's the report looking?

**Comment by natikgadzhi** [comment]
Looks like it's skipping the test on CI, but if local looks good, I'm happy to merge

**Comment by btkcodedev** [comment]
Update: Got the creds, but oauth call is getting failed with 400 bad request while testing with read command

**Comment by btkcodedev** [comment]
Update: Problems with backward compatibility of loading JSON Schemas are fixed via migrating to inline schemas,
Used `airbyte-ci connectors --name source-quickbooks migrate-to-inline_schemas`

Discover is now working properly
Ref:
image

**Comment by mkrawc** [comment]
@btkcodedev Can you elaborate why inline schemas are preferred? To me it clutters manifest.yml that usually is already quite big. IMO keeping schemas in json files should be preferred.

**Comment by btkcodedev** [comment]
@mkrawc inlining schemas and spec Is in the roadmap for making it compatible with builder.
What is your suggestion? Should we remove it?
Cc: @natikgadzhi

File: airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 version: "0.34.0"
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml:1]: The platform actually actively tries to run the CDK that is defined by this version, so it should in general match what you have in pyproject.toml

Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml:1]:
```suggestion
version: "1.0.0"
```

**Review by natikgadzhi — APPROVED** [review]

**Comment by natikgadzhi** [comment]
@btkcodedev approved with a nit — let's bump the manifest.version and merge.

**Comment by Loky1s** [comment]
R you trying to help me get my money back?????   R u helping  them take all of my money???

**Pull request merged**